app/src/main/java/fr/free/nrw/commons/upload/ReadEXIF.java

```diff
@@ -0,0 +1,46 @@
+package fr.free.nrw.commons.upload;
+
+import com.drew.imaging.ImageMetadataReader;
+import com.drew.imaging.ImageProcessingException;
+import com.drew.metadata.Directory;
+import com.drew.metadata.Metadata;
+
+import java.io.File;
+import java.io.IOException;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+import fr.free.nrw.commons.utils.ImageUtils;
+import io.reactivex.Single;
+import timber.log.Timber;
+
+@Singleton
+public class ReadEXIF {
```

Comment:
Any particular reason for using a verb rather than a noun (which is the convention for Java classes)?
I would suggest `EXIFReader`.